### ISSUE TITLE
Is this code supposed to work on the toy dataset yet ?

### ISSUE THREAD

**Issue by w2ex** [ojus1/Time2Vec-PyTorch#1]
Hello,

First of all, thank you for writing and sharing this work.
I am willing to replicate the results of the paper. To make sure everything is working as it should, I have been trying to replicate the figure 4 of the paper to see if the model is indeed learning to identify the period of 7 on the toy dataset.

However, despite trying several values for the learning rate and other few modifications, I cannot obtain anything similar to the Figure 4b of the paper ; after training, all I get is a cloud of points similar to Figure 4a.

Have you been able to discover the expected frequency on the toy dataset with your code ? Am I missing something ?
Thank you very much,

Thomas

**Comment by Filos1992** [comment]
Hey, you closed this topic, but how did you solved it?
I also try to reproduce Figure 4a and 4b but the weights don't seem to change this much.
I try to plot it with 
```
import matplotlib.pyplot as plt
b = self.model.l1.b.data.numpy()
w = self.model.l1.w.data.numpy()
plt.scatter(w,b)
plt.show()
```
But right now without luck

**Comment by w2ex** [comment]
I went in another direction, so I can't help you much.
However, you should not try to plot b (the bias in the sine function) as a function of w (the frequency).
You probably want to plot the weight of the last fully connected layer as a function of the frequency.
It should come out like this :
```
w = self.model.l1.w.data.numpy()
a = [abs(w) for w in pipe.model.fc1.weight.tolist()[0][:-1]]
plt.scatter(w,a)
plt.show()
```
Note that the [0] is to get the activation for the first/positive output node. You can probably normalize these values by the weights of the negative node. The [:-1] is to get rid off the linear component (corresponding to v2 in periodic_activations.py).

Hope this helps !

**Comment by Filos1992** [comment]
Oh thank you, 
to use the bias as the y axis was dumb, your right.
i will keep trying to reproduce the graph.
If I am allowed to ask, in which direction do you went?
did you found another solution for time series embedding?

**Comment by w2ex** [comment]
My problem implied to deal with events occurring on a large range of frequencies (from a scale of one day to a scale of one year). However, I found out that this Time2vec method was not enough to deal with a range of frequencies this large. It is really difficult to discover new frequencies lying far from the initialization values. One workaround was to set manually the initialization values, but in the end I just couldn't get the results I expected, so I gave up.
Unfortunately, I could not find another solution for time series embedding yet.
